### PR TITLE
[FIX] pos_coupon:  Inheritance in constructor of Order Model.

### PR DESCRIPTION
**Pull request by hugho-ad**
- Order initialize method is returning always the object `this` but
modules like
https://github.com/odoo/odoo/blob/114968c4a8e46b6b9e8df12b32eb4df88e57307c/addons/pos_hr/static/src/js/models.js#L83
https://github.com/odoo/odoo/blob/114968c4a8e46b6b9e8df12b32eb4df88e57307c/addons/pos_restaurant/static/src/js/floors.js#L51

Does not return them, so instead of using the result of the super, better lean on `this` object that ensure access to
object and its properties.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hugho-ad** [comment]
@trj-odoo @caburj friendly remminder

File: addons/pos_coupon/static/src/js/coupon.js [hunk 1]
```diff
@@ -254,6 +254,7 @@ odoo.define('pos_coupon.pos', function (require) {
 
         initialize: function () {
             let res = _order_super.initialize.apply(this, arguments);
+            res = res || this;
             res.on(
```
[the remark this answers is not in the document]
Comment (hugho-ad) [reply addons/pos_coupon/static/src/js/coupon.js]: sorry that was a misunderstanding of the code

FIxed!

**Comment by caburj** [comment]
@robodoo r+
@fw-bot up to saas-15.1

**Comment by hugho-ad** [comment]
@caburj this will land also in 15.0?

**Comment by caburj** [comment]
Yes, it's in 15.0 and saas-15.1. It's not an issue starting saas-15.2 since backbone is removed there.

**Pull request closed without merging**